br/pkg/restore/pipeline_items.go

```diff
@@ -386,6 +407,12 @@ func (b *tikvSender) restoreWorker(ctx context.Context, ranges <-chan drainResul
 					return e
 				}
 				log.Info("restore batch done", rtree.ZapRanges(r.result.Ranges))
+				e = appendRangesToCheckpoint(ctx, runner, r.result)
+				if e != nil {
+					log.Error("restore batch meet error, caused by checkpoint", logutil.ShortError(e))
+					r.done()
+					return e
```

Comment:
How about annotate over this error?

Reply:
At most the error is about the transfer failure between `external storage` and `br`. This kind of error cannot occurs in other progress(when br begins to restore kvs, only TiKV nodes access the `external storage`). Therefore, if annotate over this error, the checkpoint module might be unavailable but the restore is still running. And it's terrible for users when they find the restore failed but the checkpoint module has been stopped very early. 😵‍💫